hazelcast/src/main/java/com/hazelcast/nio/Address.java

```diff
@@ -162,6 +165,7 @@ public void readData(ObjectDataInput in) throws IOException {
             byte[] address = new byte[len];
             in.readFully(address);
             host = bytesToString(address);
+            socketAddr = new InetSocketAddress(host, port);
```

Comment:
Is it intended? We will compare by host instead of IP address after reading the address.

Reply:
Good catch. It's not intended. I'll prepare a fix.